client/src/components/layout/CommentForm.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState, useEffect } from "react"
+import getCurrentUser from "../../services/getCurrentUser.js"
+
+import translateServerErrors from "../../services/translateServerErrors.js"
+import ErrorList from "./ErrorList.js"
+
+const Comment = (props) => {
+  const [comment, setComment] = useState({
+    title: "",
+    content: "",
+    userId: "",
+    tripId: "",
+  })
+  const [errors, setErrors] = useState([])
+
+  const [currentUser, setCurrentUser] = useState({
+    id: "",
+    email: "",
+    userName: "",
+  })
+  useEffect(() => {
+    getCurrentUser()
+      .then((user) => {
+        setCurrentUser(user)
+      })
+      .catch(() => {
+        setCurrentUser(null)
+      })
+  }, [])
+
+  comment.tripId = props.tripId
+  comment.userId = currentUser.id
+
+  const handleInputChange = (event) => {
+    setComment({
+      ...comment,
+      [event.currentTarget.name]: event.currentTarget.value,
+    })
+  }
+
+  const handlePost = async () => {
+    try {
+      const response = await fetch("/api/v1/comments", {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json",
+        }),
+        body: JSON.stringify(comment),
+      })
+      if (!response.ok) {
+        if (response.status == 422) {
+          const body = await response.json()
+          const newErrors = translateServerErrors(body.errors)
+          return setErrors(newErrors)
+        }
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw error
+      } else {
+        const responseBody = await response.json()
+      }
+    } catch (error) {
+      console.error(`Error in fetch: ${error.message}`)
+    }
+  }
+
+  const handleSubmit = (event) => {
+    event.preventDefault()
+    handlePost()
+    if (comment.title.trim() !== "") {
+      setComment({ ...comment, title: "", content: "" })
+    }
+    props.showTrip()
+  }
```

Comment:
A couple things:
- line 70 doesnt make sense to me? you are checking it is empty so that you can empty it?
- your `handlePost` can actually RETURN something if you like! for instance, it could return true if the post is successful, or false if unsuccessful. you could then use that to evaluate if you are going to clear your form or show errors and keep the fields with their current values 
- props.showTrip() is unecessary. You shouldnt have to make another fetch request to update the component. Instead, the fetch in your `handlePost` should be defined in your `TripShow`, and then passed down to this component. `handlePost` can then take the newly persisted comment returned from the backend and add it to the list of comments on the show page.
- `handlePost` is a poor name. something more descriptive like `postComment` would be more ideal